src/frontend/js/widgets/Dashboard/utils/dashboardRoutes.tsx

```diff
@@ -86,8 +78,21 @@ const getDashboardRouteAttribute =
   ) => {
     const isLabel = attribute === RouteAttributes.LABEL;
 
-    const message = isLabel ? DASHBOARD_ROUTE_LABELS[path] : DASHBOARD_ROUTE_PATHS[path];
-    return intl.formatMessage(message, ...options);
+    if (isLabel) {
+      return intl.formatMessage(DASHBOARD_ROUTE_LABELS[path], ...options);
+    }
+
+    let outputPath = path as string;
```

Comment:
I would just get ride of `getDashboardRouteAttribute` and have theses two functions: 
```ts
/** Get the provided dashboard route label in the active locale */
export const getDashboardRouteLabel =
  (intl: IntlShape) =>
  (
    path: DashboardPath,
    ...options: [
      values?: Record<string, PrimitiveType | FormatXMLElementFn<string, string>>,
      opts?: IntlMessageFormatOptions,
    ]
  ) => {
    return intl.formatMessage(DASHBOARD_ROUTE_LABELS[path], ...options);
  };

/** Get the provided dashboard route path in the active locale */
export const getDashboardRoutePath = (originalPath: string, params?: {} | undefined) => {
  return generatePath(originalPath, params);
};
```

We can do a bit more and remove `getDashboardRoutePath` by replacing usage of `getRoutePath` by `generatePath` everywhere.
```sh
$ git grep 'getRoutePath(' | wc -l
42
```
And also get ride of `getDashboardRouteLabel` and replace it with a classic `intl.formatMessage` everywhere.

Also, `TEACHER_DASHBOARD_ROUTE_LABELS` and `LEARNER_DASHBOARD_ROUTE_LABELS` may be renamed as `teacherRouteMessages` and `learnerRouteLabel`.
I would also move them in a dedicated`utils/routeMessages.ts`.
They're used to display route link's text and can now be separeted to the routing logic.

Reply:
Interesting, I will implement this approach regarding the functions 👌